auto.def

```diff
@@ -176,6 +176,29 @@ if {1} {
   proc yesno val {
     expr {$val ? "yes" : "no"}
   }
+
+  # Recent author timestamp of file(s) as UTC, use PACKAGE_VERSION as fallback
+  proc get-author-date {format args} {
+    set oldpwd [pwd]
+    cd $::autosetup(srcdir)
+
+    if {$args eq "" || [catch {
+        set fd [open "|git log -1 --format=%at master -- $args" r]
+        gets $fd epoch
+        close $fd}] || $epoch eq ""
+    } {
+      #user-notice "No author timestamp found, falling back to PACKAGE_VERSION..."
+      if {![set pkgdate [get-define PACKAGE_VERSION]] || [catch {
```

Comment:
I don't think `set pkgdate [get-define PACKAGE_VERSION` can ever fail. How about just using `[get define PACKAGE_VERSION]` in your `[clock scan]` command?

Reply:
I wanted to avoid nested evaluations to keep it more readable and to prevent
a potential free form scan resulting in current day start when the `-format`
constraint is not given.

But it is and readability doesn’t really differ with syntax highlighting on,
so you are right and I’ve changed it, thanks.

What do you think about an explicit `then` keyword for `if` conditions? I do
not use it, because I found only two occurrences (`autosetup/cc-lib.tcl`) in
all Tcl scripts.